File: Java/Leetcode/LinkedList/PalindromeLL.java
package com.brianprojects.Leetcode.LinkedList;

public class PalindromeLL {
    //Time Complexity: O(n) || Space Complexity: O(n)
    public boolean isPalindrome(ListNode head){
        if(head == null){return false;}

        var current = head;
        ListNode secondLL = new ListNode(current.val);
        current = current.next;

        while(current !=null){
            var second = new ListNode(current.val);
            second.next = secondLL;
            secondLL = second;
            current = current.next;
        }

        while(secondLL != null && head != null){
            if(secondLL.val != head.val){
                return false;
            }
            secondLL = secondLL.next;
            head = head.next;
        }

        return true;
    }

}

File: Python/Leetcode_Python/LinkedList/PalindromeLL.py
# Time Complexity: O(n) || Space Complexity: O(n)
from typing import Optional
from Python.Leetcode_Python.LinkedList.SinglyLL import ListNode


def isPalindrome(head: Optional[ListNode]) -> bool:
    if not head:
        return False

    stack = []

    p1 = head

    while p1 is not None:
        stack.append(p1.val)
        p1 = p1.next

    p1 = head
    while p1 is not None:
        if p1.val != stack[-1]:
            return False
        stack.pop()
        p1 = p1.next

    return True
